### PR TITLE
Remove lbrynet daemon dependency, add sd blob reflector

### PR DESCRIPTION
**Pull request by jackrobison**
-remove use of lbrynet-daemon, instead use just a lbrynet dht node and blob manager to provide the minimal functionality required

-break out use of lbrycrd from lbrynet LBRYcrdWallet, avoiding a headache when it is removed from lbrynet

-refactor returned availability and costs fields

-add reflector server that will only accept sd blobs from winning claims

-don’t take three days to start up from scratch (now ~10 seconds), and don’t make my laptop cpu break a sweat

### CODE REVIEW

File: lighthouse/updater/Availability.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    def get_availability_for_name(self, name):
+        d = self.db.runQuery("select claim_id from claims where uri=?", (name, ))
+        d.addCallback(lambda (claim_id, ): self.db.runQuery("select peers from stream_availability where claim_id=?",
+                                                            (claim_id[0], )))
+        d.addCallback(lambda (peers, ): peers[0])
+        return d
+
+    def update_stream_size(self, claim_id, sd_hash):
+        def _should_update_size(sd_hash):
+            if sd_hash:
```
Comment (jobevers) [inline lighthouse/updater/Availability.py]: I prefer an exit-early style for checks like this:

```
if not sd_hash:
    return None
d = ...
```

At the very least, it reduces the level of nesting

Comment (lyoshenka) [inline lighthouse/updater/Availability.py]: :+1:

**Comment by lyoshenka** [comment]
can you explain why lighthouse is running a reflector server now? what data is going to be sent to it and how? what happens if it misses some data (its down, it doesnt get sent anything, etc)?

**Comment by jobevers** [comment]
Jack - I ran into a few bugs while trying to get lighthouse to run under lbry-in-a-box, which is what the commits in here are from.

**Comment by lyoshenka** [comment]
@jackrobison still curious about my previous question:

> can you explain why lighthouse is running a reflector server now? what data is going to be sent to it and how? what happens if it misses some data (its down, it doesnt get sent anything, etc)?

**Comment by jackrobison** [comment]
@lyoshenka It's running a reflector server so that it doesn't need to download anything to update - updates are pushed to lighthouse rather than pulled by it. 

In addition, lighthouse used to return an estimated cost field - which was a source of woe. That depended on lighthouse having the sd blob and having an up to date exchange rate, and was subject to inaccuracy. In addition to the old cost field, there was an boolean availability field, which just meant if lighthouse did or didn't have the sd blob - which is not necessarily a good indicator of availability. This pull request replaces the estimated cost field with a stream_size field, which has an int value in bytes if lighthouse has the sd blob and None if the sd is unknown. Then, separately, availability returns a peer count from the lbrynet dht. This allows the UI to more accurately know what is and isn't available, and how much a thing will cost to download.

In the event lighthouse doesn't get an sd blob, it won't have the stream size but it'll still know it's available. Whereas before it would continue to be 'unavailable' until lighthouse got the blob.

**Comment by lyoshenka** [comment]
copied from slack:

so just to confirm, if lighthouse doesnt receive any blobs via its reflector server, it still works fine?
it won't have the right cost estimate
but it will at least have an estimate, and everything else will work?
will lighthouse attempt to download the blobs the old way if it didnt get them via reflector?

**Comment by jackrobison** [comment]
>so just to confirm, if lighthouse doesnt receive any blobs via its reflector server, it still works fine?

It just won't have a stream size to report for that name. This won't alter the `peer_count`.

> it won't have the right cost estimate but it will at least have an estimate, and everything else will work?

In the case of a blank `stream_size`, when the client calls `get_est_cost` in the lbrynet api they'll be passing only a `name` parameter - which means lbrynet daemon will try to download the sd blob on its own. When given a `size` parameter it won't need to download anything. In the event that lighthouse doesn't have the size and it locally cannot be determined, then the cost is an estimate based just on a key fee (if there is one).

>will lighthouse attempt to download the blobs the old way if it didnt get them via reflector?

No, it doesn't run a full lbrynet session, only a dht node.

**Pull request merged**